### PR TITLE
3x3 HvH game on Sinatra

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@jsuchy @felipesere

Happy Easter! :rabbit: 

I created a human vs human game with a 3x3 board for the web using Sinatra.

The game loop works as before for the console game but for the web game, the game loop will only run as long as the player is ready (i.e. has a move available). 

I clear the player's move after it was made, so the player is not ready anymore. In that case the `play` method runs to the end and returns a board. That is how the web controller knows the state of the current game.

Looking forward to your feedback!

Here's a screenshot of the front end (grey boxes change to pink on hover):

ttt-front-end

### CODE REVIEW

File: lib/game_runner.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def start
       loop do
         game_option = ui.menu(GAME_OPTIONS)
         player1, player2 = player_factory.create_players(game_option)
-        game.play(player1, player2)
+        game.play([player1, player2], Board.new)
```
Comment (felipesere) [inline lib/game_runner.rb]: If you put player1 and player2 back into a list anyway, you don't need to the 
`player1, player2 = player_factory...` thing. 
Just use the array and call the variable `players`.

File: lib/human_web_player.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def make_move(board)
+    new_board = board.add_mark(@move, mark)
+    @move = nil
+    new_board
+  end
+
+  def ready?
+    @move != nil
```
Comment (felipesere) [inline lib/human_web_player.rb]: You can use `@move.nil?`

File: lib/views/index.haml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  %head
+    %title Tic Tac Toe
+    %link{:href => url("/styles.css"), :rel => "stylesheet", :type => "text/css"}
+
+  %body
+  %h1 Tic Tac Toe
+  .board
+    - @board.rows.each do |row|
+      .row
+        .outer-cell
```
Comment (felipesere) [inline lib/views/index.haml]: Why not simply use `cell`? there is no `inner-cell` class...

File: lib/views/index.haml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          - if @board.available_positions.include? row[0]
+            %a.cell{:href => url("/move?move=#{row[0]}")}
+              .empty-cell
```
Comment (felipesere) [inline lib/views/index.haml]: this could then be `.empty`...

File: lib/views/index.haml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                &nbsp;
+          - else
+            .cell.full-cell
```
Comment (felipesere) [inline lib/views/index.haml]: ...and this `full`.

File: lib/human_console_player.rb [hunk 6]
```diff
@@ -7,6 +7,10 @@ def initialize(mark, ui)
     @ui = ui
   end
 
+  def ready?
```
Comment (gemcfadyen) [inline lib/human_console_player.rb]: This ready? method returning true is duplicated in the HumanConsolePlayer and the computer_player. Could it be shared somehow?

File: lib/game_runner.rb (hunk 1 above)
Comment (RabeaGleissner) [inline lib/game_runner.rb]: Ah yes, of course! Thanks, I've changed it.

File: lib/human_web_player.rb (hunk 2 above)
Comment (RabeaGleissner) [inline lib/human_web_player.rb]: Done!

File: lib/views/index.haml (hunk 3 above)
Comment (RabeaGleissner) [inline lib/views/index.haml]: OK, makes sense. I changed all my class names according to your suggestions.

File: lib/human_console_player.rb (hunk 6 above)
Comment (RabeaGleissner) [inline lib/human_console_player.rb]: Thanks Georgina! I've already got a yellow mark for duplication in my "table of improvements". Glad you caught this one ahead of my IPM :wink:

**Pull request merged**